### PR TITLE
ui: Teams and members (PROJQUAY-4569)

### PR DESCRIPTION
**Pull request by harishsurf**
This PR adds Teams and membership tab. Following jiras are covered: 
[PROJQUAY-4569](https://issues.redhat.com/browse/PROJQUAY-4569)
[PROJQUAY-5686](https://issues.redhat.com/browse/PROJQUAY-5686)
[PROJQUAY-5687](https://issues.redhat.com/browse/PROJQUAY-5687)
[PROJQUAY-5799](https://issues.redhat.com//browse/PROJQUAY-5799)

Todo: Create team wizard ([PROJQUAY-5685](https://issues.redhat.com//browse/PROJQUAY-5685)) already exists as part of default permissions tab (#1856) which will be added once the PR is merged

### CODE REVIEW

File: web/src/resources/MembersResource.ts [hunk 1]
```diff
@@ -23,9 +22,46 @@ export async function fetchAllMembers(orgnames: string[], signal: AbortSignal) {
...
+  const teamMembersResponse = await axios.get(teamMemberUrl, {signal});
+  assertHttpCode(teamMembersResponse.status, 200);
+  return teamMembersResponse.data?.members;
+}
+
+export async function deleteTeamMemberForOrg(orgName: string, teamName: string, memberName: string) {
+  try {
+    await axios.delete(`/api/v1/organization/${orgName}/team/${teamName}/members/${memberName}`);
+  } catch (err) {
+    console.error(`Unable to delete member for team: ${teamName}`, err);
```
Comment (bcaton85) [inline web/src/resources/MembersResource.ts:57]: This swallows the error if something goes wrong with the request. React-query won't be able to catch this.

File: web/src/hooks/useTeams.ts [hunk 2]
```diff
@@ -28,3 +37,106 @@ interface createNewTeamForNamespaceParams {
...
+    page,
+    setPage,
+    perPage,
+    setPerPage,
+    search,
+    setSearch,
+  };
+}
+
+export function useDeleteTeam({orgName, onSuccess, onError}) {
```
Comment (bcaton85) [inline web/src/hooks/useTeams.ts:99]: I like how each hook is separated within the same file. We should keep this pattern.

File: web/src/resources/MembersResource.ts [hunk 3]
```diff
@@ -23,9 +22,46 @@ export async function fetchAllMembers(orgnames: string[], signal: AbortSignal) {
...
+  }
+}
+
+export async function deleteCollaboratorForOrg(orgName: string, collaborator: string) {
+  try {
+    await axios.delete(`/api/v1/organization/${orgName}/members/${collaborator}`);
+  } catch (err) {
+    console.error(`Unable to delete collaborator for org: ${orgName}`, err);
+  }
+}
```
Comment (bcaton85) [inline web/src/resources/MembersResource.ts:67]: needs newline

File: web/src/resources/TeamResources.ts [hunk 4]
```diff
@@ -25,3 +37,59 @@ export async function updateTeamForRobot(
...
+}
+
+
+export async function bulkDeleteTeams(orgName: string, teams: ITeams[]) {
+  const responses = await Promise.allSettled(
+    teams.map((team) => deleteTeamForOrg(orgName, team.name)),
+  );
+
+  // Aggregate failed responses
+  const errResponses = responses.filter(
```
Comment (bcaton85) [inline web/src/resources/TeamResources.ts:80]: Can we use [throwIfError](https://github.com/quay/quay/blob/master/web/src/resources/ErrorHandling.ts#L31-L46) here?

File: web/src/resources/TeamResources.ts [hunk 5]
```diff
@@ -25,3 +37,59 @@ export async function updateTeamForRobot(
...
+    const bulkDeleteError = new BulkOperationError<TeamDeleteError>(
+      'error deleting teams',
+    );
+    for (const response of errResponses) {
+      const reason = response.reason as TeamDeleteError;
+      bulkDeleteError.addError(reason.team, reason);
+    }
+    throw bulkDeleteError;
+  }
+}
```
Comment (bcaton85) [inline web/src/resources/TeamResources.ts:95]: newline

File: web/src/routes/OrganizationsList/Organization/Tabs/TeamsAndMembership/CollaboratorsView/CollaboratorsViewToolbar.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    page: number;
+    setPage: (page: number) => void;
+    perPage: number;
+    setPerPage: (perPage: number) => void;
+    searchOptions: string[];
+    search: SearchState;
+    setSearch: (search: SearchState) => void;
+    children?: React.ReactNode;
+  }
+
```
Comment (bcaton85) [inline web/src/routes/OrganizationsList/Organization/Tabs/TeamsAndMembership/CollaboratorsView/CollaboratorsViewToolbar.tsx:80]: newline

File: web/src/routes/OrganizationsList/Organization/Tabs/TeamsAndMembership/TeamsView/ManageMembers/ManageMembersToolbar.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    page: number;
+    setPage: (page: number) => void;
+    perPage: number;
+    setPerPage: (perPage: number) => void;
+    searchOptions: string[];
+    search: SearchState;
+    setSearch: (search: SearchState) => void;
+    children?: React.ReactNode;
+  }
+
```
Comment (bcaton85) [inline web/src/routes/OrganizationsList/Organization/Tabs/TeamsAndMembership/TeamsView/ManageMembers/ManageMembersToolbar.tsx:80]: newline

**Review by bcaton85 — CHANGES_REQUESTED** [review]
Looks good. Requesting minor changes. Can we also get e2e tests?

File: web/src/resources/MembersResource.ts (hunk 1 above)
Comment (harishsurf) [reply web/src/resources/MembersResource.ts:57]: replaced with:
```
catch(error){
    throw new ResourceError('Unable to delete team', teamName, error);
  }
  ```

**Pull request merged**